File: UniShare.API/Models/DTOs/Reviews/ReviewDto.cs
namespace UniShare.API.Models.DTOs.Reviews;

public class ReviewDto
{
    public Guid Id { get; set; }
    public Guid RentalRequestId { get; set; }
    public Guid ReviewerId { get; set; }
    public string ReviewerName { get; set; } = null!;
    public string? ReviewerAvatarUrl { get; set; }
    public int Rating { get; set; }
    public string? Comment { get; set; }
    public decimal ReputationDelta { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: UniShare.APP/lib/features/reviews/models/review_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'review_dto.g.dart';

/// Review DTO matching backend flat field shape.
///
/// Uses flat fields (reviewerId, reviewerName, etc.) instead of nested
/// UserSummaryDto objects to match the backend response.
@JsonSerializable()
class ReviewDto {
  final String id;
  final String rentalRequestId;
  final String reviewerId;
  final String reviewerName;
  final String? reviewerAvatarUrl;
  final int rating;
  final String? comment;
  final double reputationDelta;
  final DateTime createdAt;

  const ReviewDto({
    required this.id,
    required this.rentalRequestId,
    required this.reviewerId,
    required this.reviewerName,
    this.reviewerAvatarUrl,
    required this.rating,
    this.comment,
    required this.reputationDelta,
    required this.createdAt,
  });

  factory ReviewDto.fromJson(Map<String, dynamic> json) =>
      _$ReviewDtoFromJson(json);

  Map<String, dynamic> toJson() => _$ReviewDtoToJson(this);
}
